packages/client/src/usePhaserLayer.ts

```diff
@@ -0,0 +1,106 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { NetworkLayer } from "./layers/network";
+import { createPhaserLayer, PhaserLayer } from "./layers/phaser";
+import { phaserConfig } from "./layers/phaser/config";
+import useResizeObserver, { ResizeHandler } from "use-resize-observer";
+import { throttle } from "lodash";
+
+// TODO: keep+pause the old phaser instance when spinning up a new one to avoid flash?
+
+const createContainer = () => {
+  const container = document.createElement("div");
+  container.style.width = "100%";
+  container.style.height = "100%";
+  container.style.pointerEvents = "all";
+  return container;
+};
+
+type Props = {
+  networkLayer: NetworkLayer | null;
+  containerId?: string;
+  hidden?: boolean;
+};
+
+export const usePhaserLayer = ({ networkLayer, hidden = false }: Props) => {
+  const parentRef = useRef<HTMLElement | null>(null);
+  const [value, setValue] = useState<{ phaserLayer: PhaserLayer; container: HTMLElement } | null>(null);
+  const [{ width, height }, setSize] = useState({ width: 0, height: 0 });
+
+  useEffect(() => {
+    if (!networkLayer) return;
+    console.log("got new network layer");
+
+    console.log("creating phaser layer");
+    const container = createContainer();
+    if (parentRef.current) {
+      parentRef.current.appendChild(container);
+    }
+    const phaserLayerPromise = createPhaserLayer(networkLayer, {
+      ...phaserConfig,
+      scale: {
+        ...phaserConfig.scale,
+        parent: container,
+        // Phaser's default resize handling isn't great, so we'll do our own
+        // TODO: make a Phaser PR for this?
+        mode: Phaser.Scale.NONE,
+        width,
+        height,
+      },
+    });
+    phaserLayerPromise.then((phaserLayer) => setValue({ phaserLayer, container }));
```

Comment:
may need a promise ref to ignore any slow-to-resolve promises that resolve after the current promise